Add examples for companion agent and personalized tutor agent

Itz-Antaripa opened the pull request:
*Description of changes:*

Added two examples:

- Companion Agent
- Personalized Tutor Agent

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

akshseh commented:
2 comments for feedback:

1. We should not create a therapist agent. We need to restructure it. 
2. I created a [PR#5](https://github.com/strands-agents/samples/pull/5). I like `14-memory-persistent-agents` naming better so I'll adopt that as well in my PR. Lets ensure we don't have conflict.

Itz-Antaripa commented:
> 1. We should not create a therapist agent. We need to restructure it.
Renamed the agent and updated the content inside. Let me know if other changes are needed
